builtin/providers/ns1/provider.go

```diff
@@ -49,22 +44,13 @@ func Provider() terraform.ResourceProvider {
 }
 
 func ns1Configure(d *schema.ResourceData) (interface{}, error) {
-	httpClient := &http.Client{}
-	decos := []func(*ns1.Client){}
-	decos = append(decos, ns1.SetAPIKey(d.Get("apikey").(string)))
-	if v, ok := d.GetOk("endpoint"); ok {
-		decos = append(decos, ns1.SetEndpoint(v.(string)))
-	}
-	if _, ok := d.GetOk("ignore_ssl"); ok {
-		tr := &http.Transport{
-			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
-		}
-		httpClient.Transport = tr
+	config := Config{
+		Key:       d.Get("apikey").(string),
+		Endpoint:  d.Get("endpoint").(string),
```

Comment:
I think we  should follow the pattern of 

```
if v, ok := d.GetOk("endpoint"); ok {
  config.Endpoint = v.(string)
}
```

when we are adding optional parameters to the config block

Sound ok?